src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -1033,11 +1072,16 @@ internal override bool TypeSymbolEquals(TypeWithAnnotations type, TypeWithAnnota
                 return type.TypeSymbolEqualsCore(other, comparison);
             }
 
-            internal override bool TryForceResolveAsNullableValueType()
+            internal override void TryForceResolveAsNullableValueType()
             {
-                return Interlocked.CompareExchange(ref _resolved,
+                Interlocked.CompareExchange(ref _resolved,
                     _compilation.GetSpecialType(SpecialType.System_Nullable_T).Construct(ImmutableArray.Create(_underlying)),
-                    null) is null;
+                    null);
+            }
+
+            internal override void TryForceResolveAsNullableReferenceType()
+            {
```

Comment:
>{ [](start = 12, length = 1)

Can we assert it was not already resolved a a nullable value type when someone tries to force it as a nullable reference type? 
(and vice-versa when someone tries to force it as a nullable value type)

Reply:
> Can we assert it was not already resolved a a nullable value type when someone tries to force it as a nullable reference type? 
(and vice-versa when someone tries to force it as a nullable value type)

That would require more changes elsewhere, not just adding an assert here. I do not believe it is worth it.

---
In reply to: [276462064](https://github.com/dotnet/roslyn/pull/35048#discussion_r276462064) [](ancestors = 276462064)